### PR TITLE
Use ssh key to access remote repo

### PR DESCRIPTION
**Pull request by Vampouille**
* Clone remote repo at startup or update (git pull) existing one
* Make remote branch configurable: REMOTE_BRANCH
* Add FORCE_CLONE option to remove existing file in volume
* Switch to debian stretch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,14 +35,23 @@ sync:
...
  * `GIT_USERNAME`: git username for commit
  * `GIT_EMAIL`: git email for commit
 
 To push to a repository, these additional variables are required:
  * `REMOTE_NAME`: name of the git remote, eg `origin`
  * `REMOTE_URL`: git repository URL, eg `https://github.com/fvanderbiest/playground.git`
- * `TOKEN`: password or OAuth token (eg: [GitHub token](https://github.com/settings/tokens))
+
+Optional environment:
+ * `REMOTE_BRANCH`: Remote branch to use
```
Comment (fvanderbiest) [inline README.md]: defaults to master

File: README.md [hunk 2]
```diff
@@ -29,14 +35,23 @@ sync:
...
+ * `FORCE_CLONE`: Delete volume content before cloning remote repository
+
+To use SSH authentication to access remote repository, one of following
+variables must be set:
+ * `GIT_RSA_DEPLOY_KEY`: Private RSA key to use (eg: [GitHub deploy keys](https://developer.github.com/guides/managing-deploy-keys/))
+ * `GIT_RSA_DEPLOY_KEY_FILE`: Path to a file containning the private RSA key to use
```
Comment (fvanderbiest) [inline README.md]: containing

Comment (fvanderbiest) [reply README.md]: GIT_RSA_DEPLOY_KEY -> PRIVATE_KEY_STRING
GIT_RSA_DEPLOY_KEY_FILE -> PRIVATE_KEY_FILE ?

**Review by fvanderbiest — CHANGES_REQUESTED** [review]
typos

**Comment by fvanderbiest** [comment]
Nice patch, thanks :-)

File: scripts/entrypoint.sh [hunk 3]
```diff
@@ -1,27 +1,66 @@
...
+  git remote rm $REMOTE_NAME &> /dev/null
+  git remote add $REMOTE_NAME $REMOTE_URL
 
-    # set new url for remote
-    git remote rm $REMOTE_NAME &> /dev/null
-    git remote add $REMOTE_NAME $REMOTE_URL
+  echo "Fetch remote repo"
+  git fetch $REMOTE_NAME
+  echo "Reset to upstream state"
+  git reset --hard $REMOTE_NAME/$REMOTE_BRANCH
```
Comment (fvanderbiest) [inline scripts/entrypoint.sh:63]: + git clean -xdf

File: scripts/run.sh [hunk 4]
```diff
@@ -28,8 +28,8 @@ do
     msg=`eval $GIT_COMMIT_MESSAGE`
     git commit -m "${msg:-"no commit message"}"
 
-    if [ $REMOTE_NAME ]  && [ $REMOTE_URL ]  && [ $TOKEN ]; then
+    if [ $REMOTE_NAME ] && [ $REMOTE_URL ]; then
         # push to repository in the background
-        git push --force $REMOTE_NAME master &
+        git push --force $REMOTE_NAME $REMOTE_BRANCH &
```
Comment (fvanderbiest) [inline scripts/run.sh]: remove --force

**Comment by fvanderbiest** [comment]
To be double thought: geoserver image responsible of cloning / updating the datadir from upstream repo ?

File: README.md (hunk 2 above)
Comment (Vampouille) [reply README.md]: I would prefer to keep 'RSA' in the var name. What do you think about : 
* GIT_RSA_DEPLOY_KEY -> RSA_PRIVATE_KEY_STRING
* GIT_RSA_DEPLOY_KEY_FILE -> RSA_PRIVATE_KEY_FILE

File: scripts/entrypoint.sh (hunk 3 above)
Comment (Vampouille) [reply scripts/entrypoint.sh:63]: I added a `git clean` just after `git reset`. I think we need git reset *and* git clean commands.

Comment (fvanderbiest) [reply scripts/entrypoint.sh:63]: True !

**Pull request merged**